src/rgen/Microsoft.Macios.Generator/DataModel/ReturnType.cs

```diff
@@ -47,6 +49,33 @@ namespace Microsoft.Macios.Generator.DataModel;
 	/// </summary>
 	public bool IsVoid { get; }
 
+	readonly bool isNSObject = false;
+
+	public bool IsNSObject {
+		get => isNSObject;
+		init => isNSObject = value;
+	}
+
+	readonly bool isNativeObject = false;
+
+	public bool IsNativeObject {
+		get => isNativeObject;
+		init => isNativeObject = value;
+	}
```

Comment:
Both `INativeObject` and `NativeObject` exist, so these should be named `isINativeObject` and `IsINativeObject` to avoid confusion.

Reply:
Should we make a diff between them? Or just keeping track of INative is enough?

Reply:
Just tracking INativeObject is enough I think.